src/control/lib/control/request.go

```diff
@@ -92,15 +92,19 @@ func (r *request) SetSystem(name string) {
 
 // getSystem returns the system name set on the request or that returned by the
 // supplied sysGetter implementation or the build defined default name.
+//
+// NB: As of DAOS 2.2.0, the version is appended
+// to the system name in order to validate component
+// interoperability.
 func (r *request) getSystem(getter sysGetter) string {
```

Comment:
I considered that. It would be clearer, but on the other hand it bloats the messages even more. I think I'd like to go forward with this approach unless you see a serious downside to it. We can always add version later, or find some other scheme to handle versioning.

One benefit of using the system field this way is that it guarantees that newer components can't talk to older servers without versioning support, because the system name check will fail (e.g. newer agent/dmg will send `system_foo-2.2.0` and that won't match `system_foo` on the server, so the request will be rejected).

Reply:
> this would have to be backported to 2.0 in order to support 2.0-2.2 compatibility, right?

Nope. We don't guarantee that 2.2 components should be able to talk to 2.0 servers.

Reply:
> But I think either way, the request needs a getVersion() method to make the world simpler for the callers using it.

Yes, this is a good idea. I'll incorporate that in the next push.